spark/src/main/resources/error/delta-error-classes.json

```diff
@@ -2318,6 +2318,20 @@
     ],
     "sqlState" : "KD002"
   },
+  "DELTA_STREAMING_CANNOT_CONTINUE_PROCESSING_TYPE_WIDENING" : {
```

Comment:
Nit: Do we need a separate error message for this? Type widening is also a form of schema evolution. If we do, should we change it to say "data types being changed" instead of "data types being widened". I don't think this is specific to type widening (expect for those being the only type changes that we support right now).

Reply:
That helps with identifying how the schema change happened, I've leaned more on identifying it as 'type widening'
We can start with being specific about the change being a widening type change since these are the only ones we capture with this error, and update the wording in the future if we want to add more.

I'm changing the SQL conf from allowSourceTypeWidening to allowSourceTypeChange though, since changing that later would be a breaking change.